backend/app/worker.py

```diff
@@ -30,3 +35,44 @@ def update():
                 new_apps_zset[appid] = metadata.get("timestamp", 0)
         if new_apps_zset:
             db.redis_conn.zadd("new_apps_zset", new_apps_zset)
+
+
+@dramatiq.actor
+def republish_app(appid: str):
+    from .vending import VendingError
+
+    if not settings.flat_manager_build_secret or not settings.flat_manager_api:
+        return
+
+    repos = ["stable", "beta"]
+
+    # Create a token to use in the request
+    token = "Bearer " + jwt.encode(
+        {
+            "jti": "flathub_backend_internal_token",
+            "sub": "build",
+            "scope": ["republish"],
+            "apps": [appid],
+            "repos": repos,
+            "iat": datetime.utcnow(),
+            "exp": datetime.utcnow() + timedelta(minutes=5),
+            "name": "Backend token for internal use (republish_app)",
+        },
+        base64.b64decode(settings.flat_manager_build_secret),
+        algorithm="HS256",
+    )
+
+    with requests.Session() as session:
+        for repo in repos:
+            try:
+                response = session.post(
+                    f"{settings.flat_manager_api}/api/v1/repo/{repo}/republish",
+                    headers={"Authorization": token},
+                    json={"app": appid},
+                )
+
+                if response.status_code != 200:
+                    raise VendingError("republish failed")
```

Comment:
Might be a cool idea, to write these to postgres at some point and attatch them to the app for the dev to see